### ISSUE TITLE
fontname=None issues (TypeError: join() argument must be str or bytes, not 'NoneType')

### ISSUE THREAD

**Issue by lordmauve** [lordmauve/pgzero#53]
Originally reported by: **Steve Clement (Bitbucket: [SteveClement](https://bitbucket.org/SteveClement), GitHub: [SteveClement](https://github.com/SteveClement))**

----------------------------------------

Running the example ptext.py yields another error.

when fontname or systemfontname is set to **None** you will get the following:

```
#!python

pgzrun ptext.py
Traceback (most recent call last):
  File "/usr/local/bin/pgzrun", line 11, in <module>
    load_entry_point('pgzero===1.1s', 'console_scripts', 'pgzrun')()
  File "/usr/local/lib/python3.6/site-packages/pgzero/runner.py", line 89, in main
    PGZeroGame(mod).run()
  File "/usr/local/lib/python3.6/site-packages/pgzero/game.py", line 248, in run
    draw()
  File "ptext.py", line 128, in draw
    shadow=(1, 1)
  File "/usr/local/lib/python3.6/site-packages/pgzero/screen.py", line 81, in text
    ptext.draw(*args, surf=self._surf, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 442, in draw
    lineheight, angle, cache)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 269, in getsurf
    width=width, widthem=widthem, strip=strip)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 84, in wrap
    bold, italic, underline)
  File "/usr/local/lib/python3.6/site-packages/pgzero/loaders.py", line 243, in getfont
    font = fonts.load(fontname, fontsize)
  File "/usr/local/lib/python3.6/site-packages/pgzero/loaders.py", line 118, in load
    p = os.path.join(self._root(), name)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/posixpath.py", line 92, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/genericpath.py", line 149, in _check_arg_types
    (funcname, s.__class__.__name__)) from None
TypeError: join() argument must be str or bytes, not 'NoneType'
```

I get a **different** error with this code, both might be related:

```
#!python

def draw():
    screen.draw.text(
                ': {0}'.format(score),
                center=(50,50),
                # giving the correct sysfont works.
                # sysfontname='arial',
                fontsize=48
            )
```

Error:

```
#!python

$ python3 -m pgzero 012_1stText.py
👽 might win
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.6/site-packages/pgzero/__main__.py", line 3, in <module>
    main()
  File "/usr/local/lib/python3.6/site-packages/pgzero/runner.py", line 89, in main
    PGZeroGame(mod).run()
  File "/usr/local/lib/python3.6/site-packages/pgzero/game.py", line 248, in run
    draw()
  File "012_1stText.py", line 26, in draw
    fontname="arial",
  File "/usr/local/lib/python3.6/site-packages/pgzero/screen.py", line 81, in text
    ptext.draw(*args, surf=self._surf, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 442, in draw
    lineheight, angle, cache)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 269, in getsurf
    width=width, widthem=widthem, strip=strip)
  File "/usr/local/lib/python3.6/site-packages/pgzero/ptext.py", line 84, in wrap
    bold, italic, underline)
  File "/usr/local/lib/python3.6/site-packages/pgzero/loaders.py", line 243, in getfont
    font = fonts.load(fontname, fontsize)
  File "/usr/local/lib/python3.6/site-packages/pgzero/loaders.py", line 117, in load
    self.validate_root(name)
  File "/usr/local/lib/python3.6/site-packages/pgzero/loaders.py", line 99, in validate_root
    subpath=self.subpath, type=self.TYPE, name=name
KeyError: "No 'fonts' directory found to load font 'None'."
```

It looks as if None is being converted to a string somewhere.

----------------------------------------
- Bitbucket: https://bitbucket.org/lordmauve/pgzero/issue/53

**Comment by MarkAColes** [comment]
I am working on this

**Comment by MarkAColes** [comment]
I have tested the code outside of the ptext.py example and also tested the ptext.py example and am unable to replicate the bug that has been reported.  Tested using basic testing file on windows 10, run in debug on windows 10 with same test file which has the code from above to replicate the same issue and unable to do so.
Also tested ptext.py on Linux and Mac which where both unable to replicate the error reported.

I have also tried to revert to older versions of numpy and pygame but was unable to do so as they are no longer available.
